sql-to-dbsp-compiler/lib/sqllib/src/variant.rs

```diff
@@ -143,6 +146,31 @@ pub enum Variant {
     //Struct(Option<Box<dyn StructVariant>>),
 }
 
+fn json_string_escape(src: &str, buffer: &mut String) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
is this correct? serde doesn't have an utility function for this already?

Reply:
I have no good ways of answering these questions. If it's a short function whose spec is clear I don't have a problem having a copy. This does match the JSON string quoting spec https://www.json.org/json-en.html